Add utility functions for PPI analysis and functional enrichment visualization

Hinna0818 opened the pull request:
PPI_utils.R now includes 'PPI_subset()' and 'getPieData()' functions:
'PPI_subset()' : Subsets STRING PPI network data based on a selected gene list and confidence score. 
'getpiedata()': Generates summary data for pie chart visualization based on PPI results. 

gglollipop.R now includes 'gglollipop()':
'gglollipop()' : Generates lollipop plots for enrichment results.

Hinna0818 committed:
add getpieData.R

Hinna0818 committed:
Delete inst/utils/getpieData.R

Hinna0818 committed:
add PPI_utils.R

Hinna0818 committed:
add gglollipop.R

Hinna0818 committed:
Update PPI_utils.R

xiangpin commented on inst/utils/getpieData.R:
```diff
@@ -0,0 +1,65 @@
+  
+  # 4. filter the nodes in ppi
+  pie_data <- score_df %>%
+    dplyr::filter(name %in% ppi_genes)
```
tidyverse系统封装在函数中时，通常需要.data$name 以显式地指明这些名字是在数据中寻找的，而不是在函数的环境中。

xiangpin commented on R/PPI_utils.R:
```diff
@@ -0,0 +1,116 @@
+
+ppi_subset <- function(ppi_obj, n = NULL, score_cutoff = 0.7) {
+  
+  library(igraph)
```
在R包中的函数里不要library，而是相应的在上面的Rd中用`#'@importFrom igraph degree induced_subgraph`这种写法，然后在`DESCRIPITION`文件中的`import`里去声明`igraph`等相应你要导入的包

xiangpin commented on R/PPI_utils.R:
```diff
@@ -0,0 +1,116 @@
+#' @param score_cutoff Minimum edge score to keep. Default is 0.7.
+#'
+#' @return A subgraph of the original PPI network
+
```
如果函数是要公开被调用的，需要加`#'@export`

xiangpin commented on R/PPI_utils.R:
```diff
@@ -0,0 +1,116 @@
+    stop("Edges must have a 'score' attribute.")
+  }
+  
+  ppi_filtered <- subgraph.edges(ppi_obj, eids = E(ppi_obj)[score >= score_cutoff], delete.vertices = TRUE)
```
这个score是哪个向量？前面的`E(ppi_obj)$score`？没有定义的话会报错的

xiangpin commented on R/PPI_utils.R:
```diff
@@ -0,0 +1,116 @@
+  
+  # 1. extract top n pathways
+  enrich_df <- enrich_obj@result %>%
+    dplyr::arrange(p.adjust) %>%
```
同样的，用`.data$p.adjust`去写

xiangpin commented on R/PPI_utils.R:
```diff
@@ -0,0 +1,116 @@
+#' @return A data frame with genes in rows and selected enrichment terms in columns. Values represent either binary membership or weighted scores.
+#'
+
+getPieData <- function(
```
这个函数跟`getPieData.R`是重复的，只保留一个即可。同样的，这个函数如果要被公开使用就在上面的`Rd`里用`#'@export`声明

xiangpin commented on R/gglollipop.R:
```diff
@@ -0,0 +1,84 @@
+    plot_title = NULL) 
+  {
+  
+  library(ggplot2)
```
也是一样的，不用用library

xiangpin commented on R/gglollipop.R:
```diff
@@ -0,0 +1,84 @@
+    df,
+    aes(RichFactor, stats::reorder(Description, RichFactor))
+  ) +
+    geom_segment(aes(xend = 0, yend = Description)) +
```
ggplot2语法封装进函数里，也是不能直接用交互式的语法，参考https://github.com/YuLab-SMU/SVP/blob/main/R/methods-plot.R#L155

xiangpin commented on R/gglollipop.R:
```diff
@@ -0,0 +1,84 @@
+  }
+  
+  df <- df %>%
+    arrange(p.adjust) %>%
```
tidyverse框架，在函数内部的使用`.data$p.adjust`，其他地方我没有发现的也得自己调整

xiangpin commented on R/gglollipop.R:
```diff
@@ -0,0 +1,84 @@
+  
+  df$RichFactor <- df$RichFactor %>% round(digits = 2)
+  
+  label_wrap <- function(labels) str_wrap(labels, width = text.width)
```
这个`str_wrap`用`yulab.utils::str_wrap`代替，`stringr`软件包太多依赖，有些平台很难安装下载，开发时尽量考虑不调用它，而是用`base`一些函数自己写一些基础功能

xiangpin commented on R/gglollipop.R:
```diff
@@ -0,0 +1,84 @@
+  
+  p <- ggplot(
+    df,
+    aes(RichFactor, stats::reorder(Description, RichFactor))
```
这个写法也是不行的，`RichFactor`参考下面的，`stats::reorder(Description, RichFactor)参考`https://github.com/YuLab-SMU/ggtree/blob/devel/R/geom_tiplab.R#L162-L170

Hinna0818 committed:
add new files

Hinna0818 committed:
Create test.R

Hinna0818 committed:
add new data

Hinna0818 committed:
Delete data/test.R

Hinna0818 committed:
add new file

Hinna0818 committed:
add new file

Hinna0818 committed:
add ggupset_plot.R

Hinna0818 committed:
change stroke color and size for ggvenn_plot

Hinna0818 committed:
add new params into gglollipop

Hinna0818 committed:
add a DN-control degs vector

Hinna0818 committed:
add the description of DN_data

Hinna0818 committed:
rename

Hinna0818 committed:
fixed warning

Hinna0818 committed:
add new file

Hinna0818 committed:
add new file

Hinna0818 committed:
Create upset_plot.R

GuangchuangYu commented:
不要到处copy-paste代码，像最新的这个upsetplot，调用aplotExtra即可。

Hinna0818 committed:
Delete R/upset_plot.R

Hinna0818 committed:
Delete R/ggupset_plot.R

GuangchuangYu commented:
其它代码，也检查确认，尽量调用，不要copy。

如果是没有export的函数，可以用`getFromNamespace()`。

Hinna0818 committed:
Update DESCRIPTION

Hinna0818 committed:
Update PPI_utils.R

Hinna0818 committed:
Update gglollipop.R

Hinna0818 committed:
Update ggvenn_plot.R

Hinna0818 committed:
Update getvenndata.R

Hinna0818 committed:
Update gglollipop.R

Hinna0818 committed:
Update gglollipop.R

GuangchuangYu commented:
lollipop图，用`enrichplot::dotplot()`就可以了，再加上一层`geom_segment()`，再用`p$layers <- rev(p$layers)`，把线图搞成底层，需要调颜色等细节再+scale_()去弄。

代码尽量简单，能调用我们已有代码的尽量调用，这样方便维护。

Hinna0818 committed:
Update DESCRIPTION

Hinna0818 committed:
Update gglollipop.R

Hinna0818 committed:
Update gglollipop.R

Hinna0818 committed:
Delete data/venn_data.rda

Hinna0818 committed:
update venn_data

Hinna0818 committed:
Update gglollipop.R

Hinna0818 commented:
lollipop已进一步使用enrichplot完善

GuangchuangYu commented:
其它的代码也要查一下，不要到处copy。你看我公众号，也知道以前有个人，到处copy了，要建立良好的习惯。

Hinna0818 committed:
Update TCM_sankey.R

Hinna0818 committed:
Update TCM_alluvial.R

Hinna0818 committed:
Update getvennresult.R

use 'aplotExtra:::get_all_subsets_items' as default for getting the intersection result

Commit:
update ppi_utils

Hinna0818 committed:
Delete .Rhistory

Hinna0818 commented:
> 其它的代码也要查一下，不要到处copy。你看我公众号，也知道以前有个人，到处copy了，要建立良好的习惯。

老师好，目前函数能调用的都调用了。之前是因为upset_plot还没有merge，所以就先source()源码拿来测试了，现已删除，并在新示例中改为aplotExtra::upset_plot()进行调用。

Hinna0818 committed:
add

Hinna0818 committed:
Update prepare_herb_graph.R

Hinna0818 committed:
update

Hinna0818 committed:
update

Hinna0818 committed:
update

Hinna0818 committed:
add

Hinna0818 committed:
update

Hinna0818 committed:
add demo dock data

Hinna0818 committed:
add

Hinna0818 committed:
update

Hinna0818 committed:
add go circle plot

Hinna0818 committed:
update

Hinna0818 committed:
modify legend position

Hinna0818 committed:
add

Hinna0818 committed:
modify outputs

Hinna0818 committed:
rename

Hinna0818 committed:
fixed bug

xiangpin commented on R/TCM_alluvial.R:
```diff
@@ -0,0 +1,56 @@
+#'
+#' @import ggplot2
+#' @importFrom cols4all c4a
+#' @importFrom ggsankey make_long geom_alluvial geom_alluvial_text theme_sankey
```
`ggsankey`还没有在`CRAN`上发布，只在`github`，后续用户安装可能存在问题，你已经知道用`ggalluvial`绘制桑基图了，那就可以考虑只用`ggalluvial`包，这样也减少依赖。减少依赖也是在开发过程中需要考虑的（虽然不是主要的）。

Hinna0818 replied:
目前已经删除`ggsankey`的依赖，改用`ggalluvial`绘制桑基图，对应的函数为：`TCM_sankey`。

xiangpin commented on R/dot_sankey_plot.R:
```diff
@@ -0,0 +1,340 @@
+    geom_text(
+      stat = ggalluvial::StatStratum,
+      data = function(x) dplyr::filter(x, .data$axis == "Pathway"),
+      aes(label = stringr::str_wrap(
```
这个可以用`yulab.utils::str_wrap`替换，以减少依赖。

Hinna0818 replied:
`stringr`依赖过多，目前已改用`yulab.utils::str_wrap`。

xiangpin commented:
记得进行check，我在终端中只需要进入这个包的文件夹下然后`make all`就可以进行检测
<img width="817" height="514" alt="image" src="https://github.com/user-attachments/assets/dc058801-4e5d-49a4-9d5e-0413793cd236" />
现在显示的报错信息是因为有些包没有在`DESCRIPTION`里进行声明
<img width="776" height="639" alt="image" src="https://github.com/user-attachments/assets/739ed02b-d813-4878-ae83-dc0b3f5d4835" />

Hinna0818 committed:
update

Hinna0818 committed:
update

Hinna0818 commented:
> 记得进行check，我在终端中只需要进入这个包的文件夹下然后`make all`就可以进行检测 <img alt="image" width="817" height="514" src="https://private-user-images.githubusercontent.com/17870644/501404880-dc058801-4e5d-49a4-9d5e-0413793cd236.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA1NDAyMzAsIm5iZiI6MTc2MDUzOTkzMCwicGF0aCI6Ii8xNzg3MDY0NC81MDE0MDQ4ODAtZGMwNTg4MDEtNGU1ZC00OWE0LTlkNWUtMDQxMzc5M2NkMjM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDE1VDE0NTIxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2MmVlMzkzZDdlYTllMzk4NDMyNDFkMTZhYjFjODRkMDk3OGJkY2MxNjVkOTQyZDNiZjM4ZDlmYjIxNzgwODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Ukoqz4PyUJB5H2JGMpZEpooaioejdn2agkjMI_6vePE"> 现在显示的报错信息是因为有些包没有在`DESCRIPTION`里进行声明 <img alt="image" width="776" height="639" src="https://private-user-images.githubusercontent.com/17870644/501405452-739ed02b-d813-4878-ae83-dc0b3f5d4835.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA1NDAyMzAsIm5iZiI6MTc2MDUzOTkzMCwicGF0aCI6Ii8xNzg3MDY0NC81MDE0MDU0NTItNzM5ZWQwMmItZDgxMy00ODc4LWFlODMtZGMwYjNmNWQ0ODM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDE1VDE0NTIxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiZDdiZDc2YzE4MGRlNWI1MmU3Nzc5ODViYTU3ODdhODAwZDkyZTJiOGNiN2EyODdhZDM4OGUyOGUxMTY0YTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xeqDraUWm2140Vwg5oKZ-MAFBSa3Y6LzAZ-_HbkKsMs">

目前我已经完成check，之前有一些error已经去除掉了，现在有几个notes，这个需要改吗：
<img width="445" height="87" alt="114017e7ad5919d65ed57372f697d20d" src="https://github.com/user-attachments/assets/a1802952-c000-408a-a6cb-c6cebc27cbcc" />

xiangpin commented:
>tidyverse框架，在函数内部的使用.data$p.adjust，其他地方我没有发现的也得自己调整

`notes` 一般情况下，比如变量没有定义好，得解决。我看了下，还是想之前说的，很多功能里的`tidyverse`的调用直接用变量名，这种方式直接写进功能里，后面`check`的时候是会有问题。相同的问题不应该同时出现，包括昨天说的尽量减少依赖，`stringr`里的`str_wrap`功能我一开始也说过的用`yulab.utils::str_wrap`
<img width="951" height="390" alt="image" src="https://github.com/user-attachments/assets/546ca380-5f05-4f4c-82fe-2440633fe5a5" />
<img width="828" height="504" alt="image" src="https://github.com/user-attachments/assets/de93d9ac-a517-444b-9652-1f7aa30c3fa1" />

Hinna0818 committed:
r check

Hinna0818 commented:
> > tidyverse框架，在函数内部的使用.data$p.adjust，其他地方我没有发现的也得自己调整
> 
> `notes` 一般情况下，比如变量没有定义好，得解决。我看了下，还是想之前说的，很多功能里的`tidyverse`的调用直接用变量名，这种方式直接写进功能里，后面`check`的时候是会有问题。相同的问题不应该同时出现，包括昨天说的尽量减少依赖，`stringr`里的`str_wrap`功能我一开始也说过的用`yulab.utils::str_wrap` <img alt="image" width="951" height="390" src="https://private-user-images.githubusercontent.com/17870644/501797166-546ca380-5f05-4f4c-82fe-2440633fe5a5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA1OTUzMDIsIm5iZiI6MTc2MDU5NTAwMiwicGF0aCI6Ii8xNzg3MDY0NC81MDE3OTcxNjYtNTQ2Y2EzODAtNWYwNS00ZjRjLTgyZmUtMjQ0MDYzM2ZlNWE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDE2VDA2MTAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkNWM0ZmY2MjdjNGRlZTk5N2RmYjQyMzc5YWFkZDQwMzM3ODgxM2FhNjU1ZDI3NGVjYzk2MzdlOWRjY2Y5YjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Cn3_ArRsVIMGsPgCQn8ia3KfmhyOxv9pqe9zK0TkAa0"> <img alt="image" width="828" height="504" src="https://private-user-images.githubusercontent.com/17870644/501797240-de93d9ac-a517-444b-9652-1f7aa30c3fa1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA1OTUzMDIsIm5iZiI6MTc2MDU5NTAwMiwicGF0aCI6Ii8xNzg3MDY0NC81MDE3OTcyNDAtZGU5M2Q5YWMtYTUxNy00NDRiLTk2NTItMWY3YWEzMGMzZmExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDE2VDA2MTAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3MWIyZjE4MzBjOTk5NGEwNGM3YzczY2IxOWE5OWZhNGVjM2UxNTEzODVkMWUwYTRlOGQ3ZDdmZDA4NGFhOWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WPRzcJiK1q51YgYqIdTymMvh9AgDFR-di-rNDx7pGHM">

之前有些notes还没调回来，目前已经修改了所有r check里提到的notes，包括变量名和依赖，目前没有出现相关的问题了：
<img width="1682" height="330" alt="1084cc6fcd6442aec81ecb1f46506218" src="https://github.com/user-attachments/assets/409c0907-1c86-432c-83ed-2dcbb7b37d10" />

Hinna0818 committed:
fix bugs

Hinna0818 committed:
use enricher for tcm_ora instead

Hinna0818 committed:
compute node info

Hinna0818 committed:
add new node info for PPI

Hinna0818 committed:
update

Hinna0818 committed:
add radar plot for hub nodes

Hinna0818 committed:
add seed

Hinna0818 committed:
add ppi clustering method

Hinna0818 committed:
addclusterscore

Hinna0818 committed:
add TF regulation

Hinna0818 committed:
update

GuangchuangYu commented:
这里还有一个问题就是你在一个PR里的改动太多了。review起来就有点困难。

你最好是分不同的分支，每个分支加特定的功能、提交一个PR。

这样子不同的功能，在不同的PR里，review起来就比较简单。

Hinna0818 commented:
> 这里还有一个问题就是你在一个PR里的改动太多了。review起来就有点困难。
> 
> 你最好是分不同的分支，每个分支加特定的功能、提交一个PR。
> 
> 这样子不同的功能，在不同的PR里，review起来就比较简单。

好的老师，我研究一下把不同的功能放在不同的分支里去提交PR。
